src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -234,19 +247,30 @@ export class EditDataGridPanel extends GridParentComponent {
 		};
 
 		// Setup a function for generating a promise to lookup result subsets
+		// Function will be called upon refresh.
 		this.loadDataFunction = (offset: number, count: number): Promise<{}[]> => {
 			return self.dataService.getEditRows(offset, count).then(result => {
 				if (this.dataSet) {
+					let counter = 0;
 					let gridData = result.subset.map(r => {
+						// Newly added rows will have null values displayed as empty strings.
+						// Since it is currently impossible to add empty strings via edit data
+						// this must mean the empty strings on the newly added row are null values.
+						let isNewlyAddedRow = false;
+						if (this.isInNullRow && (counter === (count - 2))) {
+							isNewlyAddedRow = true;
+						}
 						let dataWithSchema = {};
 						// skip the first column since its a number column
 						for (let i = 1; i < this.dataSet.columnDefinitions.length; i++) {
+							let isNewAddedNullCell = isNewlyAddedRow && r.cells[i - 1].displayValue === '';
```

Comment:
Why can you not use isNull here?

Reply:
We have to make sure that we are on the right row (one before the null row) first before we can replace the value, having the check done once for the whole row will optimize it a bit.

Reply:
I was referring to checking whether the cell has a null value. Cell objects have isNull on them - why are you checking for an empty string here instead of using that?

Reply:
Because the newly added row will have the null values be marked as non-null (returned from the dataservice), and their display values will be empty strings, I'm not exactly sure where this comes from, but it's definitely not from anywhere in Edit Data. This is to handle this edge case.